src/goto-instrument/dump_c_class.h

```diff
@@ -21,6 +21,120 @@ Author: Daniel Kroening, kroening@kroening.com
 
 #include <goto-programs/system_library_symbols.h>
 
+/// Used for configuring the behaviour of dump_c
+struct dump_c_configurationt final
+{
+  /// Include the function declarations in the dump
+  bool include_function_decls;
+
+  /// Include the functions in the dump
+  bool include_function_bodies;
+
+  /// Include the global declarations in the dump
+  bool include_global_decls;
+
+  /// Include the typedefs in the dump
+  bool include_typedefs;
+
+  /// Include global variable definitions in the dump
+  bool include_global_vars;
+
+  /// Include struct definitions in the dump
+  bool include_compounds;
+
+  /// Define whether to follow compunds recursively
+  bool follow_compounds;
+
+  /// Include headers type declarations are borrowed from
+  bool include_headers;
+
+  dump_c_configurationt()
+    : include_function_decls(true),
+      include_function_bodies(true),
+      include_global_decls(true),
+      include_typedefs(true),
+      include_global_vars(true),
+      include_compounds(true),
+      follow_compounds(true),
+      include_headers(false)
```

Comment:
prefer in class initializers to member initializers in constructors https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c48-prefer-in-class-initializers-to-member-initializers-in-constructors-for-constant-initializers